lib/raygun.dart

```diff
@@ -0,0 +1,226 @@
+      Pointer<G_FFIVec_String> pLines = bindings.message_lines(pMsg);
+      int lineLen = pLines.ref.len;
+      for (int j = 0; j < lineLen; j++) {
+        message.value
+            .add(pLines.ref.ptr.elementAt(j).cast<Utf8>().toDartString());
+      }
+
+      msgs.add(message);
+      bindings.message_free(pMsg);
+
+      // TODO: Metadata - Rust binding is not ready
+    }
+
+    return msgs;
+  }
+
+  // This method serves for sending and editing.
+  // If there is message sent, Warp will cover the editing part.
+  send(String conversationId, String messageId, Message messages) {
+    // Convert given values to native friendly types
+    Pointer<Int8> _convoId = conversationId.toNativeUtf8().cast<Int8>();
+    Pointer<Int8> _messageId = messageId.toNativeUtf8().cast<Int8>();
+    G_FFIVec_String _messages = messages.value.cast().elementAt(0);
+    int lines = _messages.len;
+    // Invoke and result check
+    G_FFIResult_Null result = bindings.raygun_send(
+        pRaygun, _convoId, _messageId, _messages.ptr.elementAt(0), lines);
+    if (result.error.address.toString() != "0") {
+      throw WarpException(result.error);
+    }
+  }
+
+  delete(String conversationId, [String? messageId]) {
+    // Convert given values to native friendly types
+    Pointer<Int8> _convoId = conversationId.toNativeUtf8().cast<Int8>();
+    Pointer<Int8> _messageId =
+        messageId != null ? messageId.toNativeUtf8().cast<Int8>() : nullptr;
+    // Invoke and result check
+    G_FFIResult_Null result =
+        bindings.raygun_delete(pRaygun, _convoId, _messageId);
+    if (result.error.address.toString() != "0") {
+      throw WarpException(result.error);
+    }
+  }
+
+  react(String conversationId, String messageId, ReactionState reactionState,
+      String emoji) {
+    // Convert given values to native friendly types
+    Pointer<Int8> _convoId = conversationId.toNativeUtf8().cast<Int8>();
+    Pointer<Int8> _messageId = messageId.toNativeUtf8().cast<Int8>();
+    int _reactionState = reactionState.index;
+    Pointer<Int8> _emoji = emoji.toNativeUtf8().cast<Int8>();
+    // Invoke and result check
+    G_FFIResult_Null result = bindings.raygun_react(
+        pRaygun, _convoId, _messageId, _reactionState, _emoji);
+    if (result.error.address.toString() != "0") {
+      throw WarpException(result.error);
+    }
+  }
+
+  pin(String conversationId, String messageId, PinState pinState) {
+    // Convert given values to native friendly types
+    Pointer<Int8> _convoId = conversationId.toNativeUtf8().cast<Int8>();
+    Pointer<Int8> _messageId = messageId.toNativeUtf8().cast<Int8>();
+    int _pinState = pinState.index;
+    // Invoke and result check
+    G_FFIResult_Null result =
+        bindings.raygun_pin(pRaygun, _convoId, _messageId, _pinState);
+    if (result.error.address.toString() != "0") {
+      throw WarpException(result.error);
+    }
+  }
+
+  reply(String conversationId, String messageId, Message messages) {
```

Comment:
Same as with send function

Reply:
Yes, the same manner was applied as the send method.